controller/iteration_blackbox_test.go

```diff
@@ -99,6 +99,31 @@ func (rest *TestIterationREST) TestCreateChildIteration() {
 		require.NotNil(t, created)
 		compareWithGoldenUUIDAgnostic(t, filepath.Join(rest.testDir, "create", "ok_create_child.golden.json"), created)
 	})
+	rest.T().Run("success - create child iteration with ID in request payload", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Identities(1),
+			tf.Areas(1),
+			tf.Iterations(2,
+				tf.SetIterationNames("root iteration", "child iteration"),
+				tf.PlaceIterationUnderRootIteration()))
+		name := "Sprint #21"
+		childItr := fxt.IterationByName("child iteration")
+		ci := getChildIterationPayload(&name)
+		ci.Data.ID = &childItr.ID // set ID of parent and it must be ignoed by controller
```

Comment:
I don't get why we need to ignore this ID? If the UI specifies an ID (which should be optional), then this ID should be used. If the UI specifies a wrong ID, the PK error is fine IMHO.

Reply:
Yes you are right, but we are not using the payload.ID anymore in our controller hence removed.

Reply:
Well, the moment we support other clients we eventually will have to support pre-defined IDs again, right? And then we have to remove this test as well, correct?

Reply:
not sure we will need it in future for other clients, for testing it was needed hence it was added.
But does API giving the ID in create payload to be used as PK sounds a good idea?

Reply:
> But does API giving the ID in create payload to be used as PK sounds a good idea?

Yes, why not? If the UI generates an ID, there should be no problem.

Reply:
okay, but as I have discussed with @aslakknutsen we should take what we need and ignore other stuff so IMO we should merge this PR and unblock people and then we can introduce this support to specify UUID in create call in other APIs as well if needed. WDYT ?

Reply:
Also, what will happen in the case UI sends us ID and that is not unique in our DB, we will be playing to and fro multiple times(how many time can't say) till we hit unique ID in API payload.
IMHO this is incorrect behavior.